Crash when changing landscape/portrait mode

<!--
Guidelines for submitting issues:

* Bug reports which do not fill the complete issue template will be closed.
* Please have a look at our [FAQ](https://github.com/stefan-niedermann/nextcloud-notes/blob/master/FAQ.md)
* Please search the existing issues first, it's likely that your issue was already reported or even fixed.
* This repository is *only* for issues within the Nextcloud Notes Android app
-->

<!-- Please keep this note for other contributors -->
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 


**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
App crashes when editing a note and changing from portrait to landscape mode.

**To Reproduce**
Steps to reproduce the behavior:
1.  Create a note or open an existing note in editing mode
2. Change screen orientation from portrait to landscape and/or v.v.
3. Often the app crashes to the crash screen, often closes showing the crash screen at reopening, sometimes crashes with popup "Notes keep crashing"

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
Change from portrait to landscape or v.v.

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->


**Smartphone (please complete the following information):**
 - Nextcloud Notes-Version (android app): 2.16
4
 - F-Droid or Play Store: F-droid
 - Android-Version: 10
 - Device: Samsung Galaxy A70/SM-A705FN/DS


**Server**
 - Nextcloud version: 
 - Nextcloud Notes version (server app): 

**Stacktrace**
```
App Version: 2.16.4
App Version Code: 2016004
App Flavor: fdroid

Files App Version Code: 30120190

---

OS Version: 4.14.117-17976949(A705FNXXU5BTF1)
OS API Level: 29
Device: a70q
Manufacturer: samsung
Model (and Product): SM-A705FN (a70qxx)

---

java.lang.RuntimeException: Unable to start activity ComponentInfo{it.niedermann.owncloud.notes/it.niedermann.owncloud.notes.edit.EditNoteActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.ViewGroup.removeAllViews()' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3448)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3595)
	at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5536)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5444)
	at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2147)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:7814)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1068)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.view.ViewGroup.removeAllViews()' on a null object reference
	at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:545)
	at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:166)
	at it.niedermann.owncloud.notes.edit.EditNoteActivity.onCreate(EditNoteActivity.java:54)
	at android.app.Activity.performCreate(Activity.java:7955)
	at android.app.Activity.performCreate(Activity.java:7944)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3423)
	... 13 more

```

--- a/app/src/main/java/it/niedermann/owncloud/notes/accountpicker/AccountPickerDialogFragment.java
+++ b/app/src/main/java/it/niedermann/owncloud/notes/accountpicker/AccountPickerDialogFragment.java
@@ -10,6 +10,7 @@ import android.view.WindowManager;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
 import androidx.fragment.app.DialogFragment;
 import androidx.recyclerview.widget.RecyclerView;
 
@@ -58,30 +59,32 @@ public class AccountPickerDialogFragment extends BrandedDialogFragment {
     @NonNull
     @Override
     public Dialog onCreateDialog(Bundle savedInstanceState) {
-        View view = View.inflate(getContext(), R.layout.dialog_choose_account, null);
-        DialogChooseAccountBinding binding = DialogChooseAccountBinding.bind(view);
+        final List<LocalAccount> accountsList = NotesDatabase.getInstance(getActivity()).getAccounts();
+        final AlertDialog.Builder dialogBuilder = new BrandedAlertDialogBuilder(requireActivity())
+                .setTitle(R.string.simple_move)
+                .setNegativeButton(android.R.string.cancel, null);
 
-        NotesDatabase db = NotesDatabase.getInstance(getActivity());
-        List<LocalAccount> accountsList = db.getAccounts();
+        if (accountsList.size() > 1) {
+            final DialogChooseAccountBinding binding = DialogChooseAccountBinding.inflate(LayoutInflater.from(requireContext()));
 
-        for (int i = 0; i < accountsList.size(); i++) {
-            if (accountsList.get(i).getId() == accountIdToExclude) {
-                accountsList.remove(i);
-                break;
+            for (int i = 0; i < accountsList.size(); i++) {
+                if (accountsList.get(i).getId() == accountIdToExclude) {
+                    accountsList.remove(i);
+                    break;
+                }
             }
+
+            RecyclerView.Adapter<AccountChooserViewHolder> adapter = new AccountChooserAdapter(accountsList, (account -> {
+                accountPickerListener.onAccountPicked(account);
+                dismiss();
+            }));
+            binding.accountsList.setAdapter(adapter);
+            dialogBuilder.setView(binding.getRoot());
+        } else {
+            dialogBuilder.setMessage(getString(R.string.no_other_accounts));
         }
 
-        RecyclerView.Adapter<AccountChooserViewHolder> adapter = new AccountChooserAdapter(accountsList, (account -> {
-            accountPickerListener.onAccountPicked(account);
-            dismiss();
-        }));
-        binding.accountsList.setAdapter(adapter);
-
-        return new BrandedAlertDialogBuilder(requireActivity())
-                .setView(binding.getRoot())
-                .setTitle(R.string.simple_move)
-                .setNegativeButton(android.R.string.cancel, null)
-                .create();
+        return dialogBuilder.create();
     }
 
     @Nullable

--- a/app/src/main/java/it/niedermann/owncloud/notes/edit/BaseNoteFragment.java
+++ b/app/src/main/java/it/niedermann/owncloud/notes/edit/BaseNoteFragment.java
@@ -65,7 +65,6 @@ public abstract class BaseNoteFragment extends BrandedFragment implements Catego
     private static final String SAVEDKEY_NOTE = "note";
     private static final String SAVEDKEY_ORIGINAL_NOTE = "original_note";
 
-    private boolean canMoveNoteToAnotherAccounts = false;
     private LocalAccount localAccount;
     private SingleSignOnAccount ssoAccount;
 
@@ -94,10 +93,6 @@ public abstract class BaseNoteFragment extends BrandedFragment implements Catego
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        new Thread(() -> {
-            canMoveNoteToAnotherAccounts = db.getAccountsCount() > 1;
-            requireActivity().invalidateOptionsMenu();
-        }).start();
         try {
             this.ssoAccount = SingleAccountHelper.getCurrentSingleSignOnAccount(requireActivity().getApplicationContext());
             this.localAccount = db.getLocalAccountByAccountName(ssoAccount.name);
@@ -196,7 +191,6 @@ public abstract class BaseNoteFragment extends BrandedFragment implements Catego
         if (isRequestPinShortcutSupported(requireActivity()) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
             menu.add(Menu.NONE, MENU_ID_PIN, 110, R.string.pin_to_homescreen);
         }
-        menu.findItem(R.id.menu_move).setVisible(canMoveNoteToAnotherAccounts);
 
         super.onCreateOptionsMenu(menu, inflater);
     }

--- a/app/src/main/java/it/niedermann/owncloud/notes/persistence/NotesDatabase.java
+++ b/app/src/main/java/it/niedermann/owncloud/notes/persistence/NotesDatabase.java
@@ -773,10 +773,11 @@ public class NotesDatabase extends AbstractNotesDatabase {
         return account;
     }
 
+    @NonNull
     public List<LocalAccount> getAccounts() {
         final SQLiteDatabase db = getReadableDatabase();
         final Cursor cursor = db.query(table_accounts, new String[]{key_id, key_url, key_account_name, key_username, key_etag, key_modified, key_api_version, key_color, key_text_color, key_capabilities_etag}, null, null, null, null, null);
-        final List<LocalAccount> accounts = new ArrayList<>();
+        final List<LocalAccount> accounts = new ArrayList<>(cursor.getCount());
         while (cursor.moveToNext()) {
             LocalAccount account = new LocalAccount();
             account.setId(cursor.getLong(0));

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -304,4 +304,5 @@
     <string name="error_action_open_battery_settings">Battery settings</string>
     <string name="error_action_open_deck_info">Open App info</string>
     <string name="error_action_open_network">Network settings</string>
+    <string name="no_other_accounts">You don\'t have configured any other accounts yet.</string>
 </resources>

